// Webezy.io Generated proto DO NOT EDIT
syntax = "proto3";

import "google/protobuf/struct.proto";
package webezy.WebezyCore.v1;



// [webezy.WebezyCore.v1.OkResponse] - This is OK Status response for RPC which excute a process
message OkResponse {
	// [webezy.WebezyCore.v1.OkResponse.status] - Should be OK for succssfull process
	string status = 1;
	// [webezy.WebezyCore.v1.OkResponse.context] - A human readable context for the process if any message should be more then just "OK"
	string context = 2;
}

// [webezy.WebezyCore.v1.WebezyServer] - The description for webezy server
message WebezyServer {
	// [webezy.WebezyCore.v1.WebezyServer.language] - The server language
	webezy.WebezyCore.v1.WebezyLanguage language = 1;
}

// [webezy.WebezyCore.v1.WebezyNpm] - Webezy Node Package Manager integration configurations for wrapping the Typescript / Javascript packages directly to npm with every build command
message WebezyNpm {
	// [webezy.WebezyCore.v1.WebezyNpm.scope] - The user or organization scope see: https://docs.npmjs.com/about-scopes
	string scope = 1;
	// [webezy.WebezyCore.v1.WebezyNpm.public] - If package access is public see: https://docs.npmjs.com/about-public-packages
	bool public = 2;
	// [webezy.WebezyCore.v1.WebezyNpm.token] - The npm access token see: https://docs.npmjs.com/creating-and-viewing-access-tokens
	string token = 3;
}

// [webezy.WebezyCore.v1.WebezyClient] - The description for webezy client
message WebezyClient {
	// [webezy.WebezyCore.v1.WebezyClient.language] - The client language
	webezy.WebezyCore.v1.WebezyLanguage language = 1;
	// [webezy.WebezyCore.v1.WebezyClient.out_dir] - The output path for the client code modules
	string out_dir = 2;
	// [webezy.WebezyCore.v1.WebezyClient.package] - Client package configurations for 3rd parties
	oneof package {

		webezy.WebezyCore.v1.WebezyNpm npm = 3;
	}
}

// [webezy.WebezyCore.v1.WebezyProject] - The webezy project description
message WebezyProject {
	// [webezy.WebezyCore.v1.WebezyProject.uri] - The project URI of root directory where the webezy.json file exists
	string uri = 1;
	// [webezy.WebezyCore.v1.WebezyProject.name] - Project full display name
	string name = 2;
	// [webezy.WebezyCore.v1.WebezyProject.package_name] - project name code compatiable
	string package_name = 3;
	// [webezy.WebezyCore.v1.WebezyProject.version] - The project version following SEMVER semantics
	string version = 4;
	// [webezy.WebezyCore.v1.WebezyProject.type] - The project resource type
	string type = 5;
	// [webezy.WebezyCore.v1.WebezyProject.kind] - The project resource kind
	string kind = 6;
	// [webezy.WebezyCore.v1.WebezyProject.properties] - Additional project properties
	google.protobuf.Struct properties = 7;
	// [webezy.WebezyCore.v1.WebezyProject.server] - The project attached server descriptin
	webezy.WebezyCore.v1.WebezyServer server = 8;
	// [webezy.WebezyCore.v1.WebezyProject.clients] - The project assigned clients array
	repeated webezy.WebezyCore.v1.WebezyClient clients = 9;
	// [webezy.WebezyCore.v1.WebezyProject.go_package] - The go compatiable package name
	string go_package = 10;
}

// [webezy.WebezyCore.v1.WebezyLanguage] - None
enum WebezyLanguage {
	// [webezy.WebezyCore.v1.WebezyLanguage] - None
	UNKNOWN_WEBEZYLANGUAGE = 0;
	// [webezy.WebezyCore.v1.WebezyLanguage] - None
	python = 1;
	// [webezy.WebezyCore.v1.WebezyLanguage] - None
	typescript = 2;
	// [webezy.WebezyCore.v1.WebezyLanguage] - None
	go = 3;
	// [webezy.WebezyCore.v1.WebezyLanguage] - None
	webpack = 4;
	// [webezy.WebezyCore.v1.WebezyLanguage] - None
	javascript = 5;
	// [webezy.WebezyCore.v1.WebezyLanguage] - None
	csharp = 6;
	// [webezy.WebezyCore.v1.WebezyLanguage] - None
	java = 7;
}
